### PR TITLE
DBZ-6265 skip the wal level check if no cdc is to occur

### PR DESCRIPTION
**Pull request by alwaysbemark**
https://issues.redhat.com/browse/DBZ-6265 
Refactored the Postgres connector class to extract the individual connection check methods, changed the WAL_LEVEL test to only run if the snapshot mode involves streaming.

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 1]
```diff
@@ -142,6 +100,63 @@ protected void validateConnection(Map<String, ConfigValue> configValues, Configu
         }
     }
 
+    private static void checkLoginReplicationRoles(PostgresConnection connection) throws SQLException {
```
Comment (alwaysbemark) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:103]: All methods extracted as is from the large validation method above.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 2]
```diff
@@ -142,6 +100,63 @@ protected void validateConnection(Map<String, ConfigValue> configValues, Configu
...
+                                || rs.getBoolean("aws_replication")),
+                        "Could not fetch roles"))) {
+            final String errorMessage = "Postgres roles LOGIN and REPLICATION are not assigned to user: " + connection.username();
+            LOGGER.error(errorMessage);
+        }
+    }
+
+    private static void checkWalLevel(ConfigValue hostnameValue, PostgresConnection connection, PostgresConnectorConfig config) throws SQLException {
+        // Logical WAL_LEVEL is only necessary for CDC snapshotting
+        if (config.getSnapshotter().shouldStream()) {
```
Comment (alwaysbemark) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:139]: This is the only business logic change - this check will skip for `initial_only` as it's not needed in this case.

**Comment by jpechane** [comment]
Let's discuss at https://issues.redhat.com/browse/DBZ-6271

**Comment by alwaysbemark** [comment]
> Let's discuss at https://issues.redhat.com/browse/DBZ-6271

Thanks for reviewing! This change should be safe as only checking the WAL level is skipped?

**Comment by jpechane** [comment]
@alwaysbemark Let's use the same approach as with MySQL PR. Throw error by default, log `WARN` for schema only.

**Comment by jpechane** [comment]
@alwaysbemark Applied, thanks

**Pull request merged**